### PR TITLE
fix: types

### PR DESCRIPTION
**Pull request by kellyjosephprice**
| [![PR App][icn]][demo] |
| :--------------------: |

## 🧰 Changes

Fixes a bunch of types from the jsx-coercion PR.

## 🧬 QA & Testing

- [Broken on production][prod].
- [Working in this PR app][demo].

[demo]: https://markdown-pr-PR_NUMBER.herokuapp.com
[prod]: https://SUBDOMAIN.readme.io
[icn]: https://user-images.githubusercontent.com/886627/160426047-1bee9488-305a-4145-bb2b-09d8b757d38a.svg

### CODE REVIEW

File: index.tsx
Comment (erunion) [inline index.tsx:1]: looks like there's a type error on line 96 now?

File: index.tsx [hunk 1]
```diff
@@ -112,12 +112,12 @@ export const html = (text: string, opts = {}) => {
   unimplemented('html export');
 };
 
-const astProcessor = (opts = {}) =>
+const astProcessor = (opts: MdastOpts = { components: {} }) =>
```
Comment (erunion) [inline index.tsx:115]: Can `components` get more specific or is it just `Record<string, string>`?

**Review by jennspencer — APPROVED** [review]
⌨️

(TIL there's no typewriter emoji???)

**Comment by kellyjosephprice** [comment]
> ⌨️
> 
> (TIL there's no typewriter emoji???)

https://www.unicode.org/pending/proposals.html

File: processor/transform/readme-components.ts [hunk 2]
```diff
@@ -12,88 +15,104 @@ const types = {
...
-  }, {});
+  }, {} as T);
+
+interface Options {
+  components: Record<string, string>;
+}
 
-const readmeComponents =
+const coerceJsxToMd =
   ({ components = {} } = {}) =>
```
Comment (erunion) [inline processor/transform/readme-components.ts:32]: same q here, can `components` get narrowed?

File: index.tsx (hunk 1 above)
Comment (kellyjosephprice) [reply index.tsx:115]: I don't think so? It's a mapping of tag name to component body?

File: index.tsx
Comment (kellyjosephprice) [reply index.tsx:1]: https://github.com/readmeio/markdown/commit/46c0acdf05dc6b4ee9e77a8f4ca5d24429a914e7

File: index.tsx (hunk 1 above)
Comment (erunion) [reply index.tsx:115]: might want to narrow it to `Record<string, string>` because right now it'll allow `Record<string, unknown>`

Comment (kellyjosephprice) [reply index.tsx:115]: How?

image

Comment (erunion) [reply index.tsx:115]: oh my bad. `MdastOpts` is where `components` is defined, I misread the `=` there as `:` and thought you were typing `opts: { components: {} }`

Comment (kellyjosephprice) [reply index.tsx:115]: Thought I was about to have my type system shattered.

Comment (erunion) [reply index.tsx:115]: nope im just a dumbass. carry on

**Comment by rafegoldberg** [comment]
## This PR was released!

🚀 Changes included in [<b>v6.75.0-beta.32</b>](/readmeio/markdown/releases/tag/v6.75.0-beta.32)
<br>

**Pull request merged**